overlord/snapstate/snapstate.go

```diff
@@ -1273,6 +1275,17 @@ func doUpdate(ctx context.Context, st *state.State, names []string, updates []*s
 		updated = append(updated, name)
 	}
 
+	// Report a useful error for auto-refresh if nothing can be
+	// updated and we have at least one conflicting change in
+	// progress.
+	if refreshAll && len(tasksets) == 0 {
+		for _, err := range updateErrs {
+			if _, ok := err.(*ChangeConflictError); ok {
+				return nil, nil, &ChangeConflictError{Message: "auto-refresh has conflicting changes in progress"}
```

Comment:
why auto-refresh?   if the user did "snap refresh" we get here as well.  Also thinking more, this is a potentially backward incompatible change...

Reply:
Good point, let me fix this. 

It breaks compatibility because it now exits with an exit code instead of 0, yes? So this needs special handling at least for a while to show a message but exit with zero? Or something else?

Reply:
I need to remember how all this works, but yes sorry, is not super simple :/

Reply:
snap warning maybe?

Reply:
not only the exit != 0 of the command but also the API returning an error where it typically wouldn't at all